### ISSUE TITLE
On/off and switch source doesn't work

### ISSUE THREAD

**Issue by vargylet** [toebsen/python-denonavr#8]
Hi,

Really like this simple solution, but I can't get it to properly work. Everything works except _/power/turnon_, _/power/turnoff_ and _/input/switch/<source_id>_.

This is what I get when I try to connect to /power/turnoff:

`[2019-09-22 16:06:21,351] ERROR in app: Exception on /power/turnoff [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "build/bdist.linux-x86_64/egg/denonavr/__main__.py", line 93, in turn_off
    return jsonify(power=execute("PWSTANDBY", CONFIG))
  File "build/bdist.linux-x86_64/egg/denonavr/__main__.py", line 51, in execute
    tn_con = telnetlib.Telnet(config["host"], config["port"], config["timeout"])
  File "/usr/lib/python2.7/telnetlib.py", line 211, in __init__
    self.open(host, port, timeout)
  File "/usr/lib/python2.7/telnetlib.py", line 227, in open
    self.sock = socket.create_connection((host, port), timeout)
  File "/usr/lib/python2.7/socket.py", line 575, in create_connection
    raise err
error: [Errno 111] Connection refused
10.0.1.5 - - [22/Sep/2019 16:06:21] "GET /power/turnoff HTTP/1.1" 500 -`

I'm running denonavr_server with the -c flag to my config file, the complete path. The system runs Python 2.7.13 and pip 19.2.3. I've tried turning the firewall off completely, but it doesn't help.

Any ideas?

Kind regards,
Johan

**Comment by toebsen** [comment]
Hi Johan,
Can it be that the port is already opened on the AVR somehow? Since  _connection is refused_ is the main exception. Additionally there is also a setting on the AVR networks settings, such that the network interface is always on. I guess _Network/IPControl ->  Always on_. Maybe can you provide more logs/information? Also note that I only tested it on a Denon AVR X1000

**Comment by vargylet** [comment]
Hi,

I’m sorry about my super late response! 

I haven’t had a chance to tinker with this since I wrote the initial post. But since then Apple released a new version of iOS 13 that fixed the Denon remote app. It was broken in iOS 13.0 and that’s why I started looking at your solution. Since the app itself works again I won’t try to get this to work again.

Sorry for taking up your time and energy! I’ll close this. 

Kind regards,
Johan
